src/ifp/src/InitFloorplan.tcl

```diff
@@ -263,4 +269,33 @@ proc microns_to_mfg_grid { microns } {
   }  
 }
 
+proc cut_rows {block placement_blockages} {
+  utl::info "IFP" 6 "Placement blockages found: [llength $placement_blockages]"
+
+  # Gather rows needing to be cut because of placement blockages
+  set rows_to_cut []
+  set row_placement_blockages [dict create]
+  foreach blockage $placement_blockages {
+    if {![$blockage isSoft]} {
+      foreach row [$block getRows] {
+        if {[tap::overlaps $blockage $row 0 0]} {
+          set row_name [$row getName]
+          if {![dict exists $row_placement_blockages $row_name]} {
```

Comment:
This check should move before the overlap check, and if it's already removed the loop should continue. If the row is already removed there's no benefit to checking if it overlaps.